example/src/main/java/com/dynatrace/metric/util/example/App.java

```diff
@@ -97,4 +98,24 @@ public static void main(String[] args) {
       System.out.println(me);
     }
   }
+
+  // to try this out, create the endpoint.properties file, start the app, and then modify the
+  // contents of the file.
+  void testFilePolling() {
+    // file is at /var/lib/dynatrace/enrichment/endpoint/endpoint.properties
+    final DynatraceFileBasedConfigurationProvider instance =
+        DynatraceFileBasedConfigurationProvider.getInstance();
+
+    int counter = 0;
+    while (true) {
+      System.out.println(String.format("=============== %d ===============", counter++));
+      System.out.println("Endpoint: " + instance.getMetricIngestEndpoint());
+      System.out.println("Token: " + instance.getMetricIngestToken());
```

Comment:
```suggestion
      String token = instance.getMetricIngestToken();
      System.out.println(String.format("=============== %d ===============", counter++));
      System.out.println("Endpoint: " + instance.getMetricIngestEndpoint());
      System.out.println("Token:    " + token.substring(0, Math.min(token.length(), 32))); // 32 chars = public portion only if actual token
```
You could just take the first 32 characters here (= token prefix + public portion) to avoid printing the entire token in case someone accidentally runs this on a production environment where this would make it into the logs.